docs/spec/dns-integration.md

```diff
@@ -4,23 +4,32 @@ DNS Integration for NSM
 Specification
 -------------
 
-Network Service Mesh needs to be able to provide a workload with DNS service from Network Services Without breaking K8s DNS
+Network Service Mesh needs to be able to provide a workload with DNS resolution for Network Services without breaking the DNS resolution K8s natively provides.
+
+Overview
+--------
+
+The DNS integration capability is performed by running a DNS server co-resident with the application pod that can direct requests to multiple DNS servers. The DNS servers that should be sent requests is controlled by the NSEs that create connections into that application pod.  If the NSEs populate the DNSContext with a DNS server appropriate for this connection then additional containers in the application pod will ensure those DNS servers are consulted. The additional containers are nsm-coredns and nsm-dns-monitor.  They can be inserted directly or via the admission webhook.  The nsm-coredns container is responsible for handling the DNS resolution mechanics. The nsm-dns-monitor container is responsible for updating the configuration that the nsm-coredns container acts on based on DNS information in the connection contexts provided by the NSEs.
+
+The NSM provided SDK provides a capability to read environmental varibles to determine additional DNS servers and domains that should be added to the DNScontext.  These variables are: DNS_SEARCH_DOMAINS and DNS_SERVER_IPS.
+
+Note:  Some of the additional DNS configuration capabilites K8s offers are overridden adn not availabel when these additional containers are inserted.
```

Comment:
Which ones?  We tried hard to make sure all normal K8s DNS things worked... what did we miss?

Reply:
I was a bit speculative here but some of the options described here:  https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-config almost certainly don't work.   I have not directly tested this so willing to reword this if its an issue,   but I think we should provide some warning to not mix the advanced pod DNS config with the fanout plugin NSM is implementing

Reply:
@denis-tingajkin How are we handling these cases @john-a-joyce referenced here?

Reply:
@edwarnicke At this moment we have implemented `resolv.conf` parser to reading user's nameservers and search domains. But we did not a solution for some not default [options](http://man7.org/linux/man-pages/man5/resolv.conf.5.html).

Reply:
I've checked. Options also supported. We just keep user's options from POD DNS config.

Reply:
I think, In general, we can use any K8S DNS configuration capabilities.

Reply:
I would be surprised if all the capabilities that K8s offers work as the user would expect when the fanout is used, but I also have not fully tested or evaluated. So it is best that I just remove this warning entirely at the current time.